src/main/java/dev/waterdog/waterdogpe/network/bridge/DownstreamBridge.java

```diff
@@ -40,10 +41,14 @@ public boolean handlePacket(BedrockPacket packet, BedrockPacketHandler handler)
         RewriteMaps rewriteMaps = this.player.getRewriteMaps();
         boolean rewroteBlock = rewriteMaps.getBlockMap() != null && rewriteMaps.getBlockMap().doRewrite(packet);
 
-        boolean pluginHandled = false;
-        if (this.player.getPluginDownstreamHandler() != null) {
-            pluginHandled = this.player.getPluginDownstreamHandler().handlePacket(packet);
+        AtomicBoolean pluginHandled = new AtomicBoolean(false);
```

Comment:
Using `AtomicBoolean` creates here unnecessary object allocation for no reason. It would be better to use normal `for` loop instead.